modules/dcache/src/main/java/org/dcache/pool/classic/IoQueueManager.java

```diff
@@ -451,16 +461,43 @@ public Serializable call() throws NoSuchElementException
                     comparator = comparator.reversed();
                 }
 
+                // Prepare filters
+                Predicate<MoverRequestScheduler.PrioritizedRequest> clientIpFilter;
+                if (clientHost != null && !clientHost.isEmpty()) {
+                    try {
+                        // check and resolve host
+                        clientHost = InetAddress.getByName(clientHost).getHostAddress();
+                    } catch (UnknownHostException e) {
+                        return String.format("Unknown host %s", clientHost);
+                    }
+
+                    // building the filter
+                    clientIpFilter = prioritizedRequest -> {
+                        ProtocolInfo info = prioritizedRequest.getMover().getProtocolInfo();
+                        // ip protocol only
+                        if (info instanceof IpProtocolInfo) {
+                            return ((IpProtocolInfo) info).getSocketAddress().equals(clientHost);
```

Comment:
`getSocketAddress` returns an `InetSocketAddress`, whereas `clientHost` is of type `String`.  This means this test will never succeed.

Incidentally, testing IP addresses by doing a String comparison is a somewhat risky strategy -- it probably works with IPv4 address, but an IPv6 address can be "written as a String" in different ways.